pkg/module/resources.go

```diff
@@ -0,0 +1,157 @@
+package module
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	cdv2 "github.com/gardener/component-spec/bindings-go/apis/v2"
+	"github.com/gardener/component-spec/bindings-go/apis/v2/cdutils"
+	cdvalidation "github.com/gardener/component-spec/bindings-go/apis/v2/validation"
+	"github.com/gardener/component-spec/bindings-go/ctf"
+	"github.com/kyma-project/cli/internal/files"
+	"github.com/kyma-project/cli/pkg/module/blob"
+	"github.com/mandelsoft/vfs/pkg/vfs"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"sigs.k8s.io/yaml"
+)
+
+// ResourceDescriptor contains all information to describe a resource
+type ResourceDescriptor struct {
+	cdv2.Resource
+	Input *blob.Input `json:"input,omitempty"`
+}
+
+// ResourceDescriptorList contains a list of all informations to describe a resource.
+type ResourceDescriptorList struct {
+	Resources []ResourceDescriptor `json:"resources"`
+}
+
+// AddResources adds the resources in the given resource definitions into the archive and its FS.
+// A resource definition is a string with format: NAME:TYPE@PATH where NAME and TYPE can be omitted and will default to the last path element name and "helm-chart" respectively
+func AddResources(archive *ctf.ComponentArchive, c *ComponentConfig, log *zap.SugaredLogger, fs vfs.FileSystem, resourceDefs ...string) error {
```

Comment:
maybew we should extract the resourceDef type into a custom struct with some validation

Reply:
That is a god idea yes! Will change it.

Reply:
Added a new type for resource definitions and moved all the parsing and setting defaults logic there.